### ISSUE TITLE
[Discussion] Update for more modern network calls

### ISSUE THREAD

**Issue by squarefrog** [fkuehne/upnpx#25]
**Note:** this is a discussion rather than an issue. 

Upnpx is currently pretty feature complete and rich, but is starting to show its age a little. One of the biggest components of this is the use of synchronous network calls. I understand why this was used, but I think a more modern approach would be asynchronous. 

My biggest issue with synchronous calls is how it blocks the main thread while it waits for a response. This may not be an issue with a small library on a fast local network, it becomes a huge issue when you have an extensive library on an older device such as the WD My Life hard drive running Twonky. 

The other advantage to asynchronous calls would be lazy loading table views. We could fetch the first 30 objects and add them to the data source, then further fetch batches as the user scrolls. 

Now, this is by no means a trivial task - and could create some API breaking changes, so I suppose the decision needs to be made whether it's advantages outweigh the disadvantages.

**Comment by lightbow** [comment]
-[SoapAction action:parameters:returnValues:] is calling +[NSURLConnection sendSynchronousRequest:returningResponse:error:] which Apple specifically calls out in the docs as "you should never call this function from the main thread of a GUI application."

**Comment by squarefrog** [comment]
Precisely. It's obviously why you shouldn't do this while scrolling a table view while a call is made. 

You can create a pseudo-asynchronous call by using operation queues or GCD.

**Comment by lightbow** [comment]
Thanks. I'm having good luck so far with using NSOperation subclasses to get this stuff off the main thread, encapsulating the soap actions and some post-processing, and generally making my code cleaner.
